xmodule/capa/safe_exec/tests/test_safe_exec.py

```diff
@@ -87,7 +88,11 @@ def test_cant_do_something_forbidden(self):
         '''
         # Can't test for forbiddenness if CodeJail isn't configured for python.
         if not jail_code.is_configured("python"):
-            pytest.skip()
+
+            # Can't test for forbiddenness if CodeJail rest service isn't enabled.
+            # Remote codejailservice must be running, see https://github.com/eduNEXT/tutor-contrib-codejail/
+            if not is_codejail_rest_service_enabled():
+                pytest.skip(reason="Local or remote codejail has to be configured and enabled to run this test.")
```

Comment:
@xitij2000, I don't have a tutor instance to test this, but shouldn't it be an `or` condition (as it was in 3c4b95e0dfe5ee7d8b4f7eb98b3383db0e134bfc) instead of an `and`? Currently, it's equal to `if not jail_code.is_configured("python") and not is_codejail_rest_service_enabled()`.

Reply:
It should be an AND condition since they are alternatives. Basically if neither mechanism for running codejail is configured, skip the test, otherwise run it. 

For testing without tutor, there is a way to do that. 

1. Disable regular codejail. You can do [what tutor does](https://github.com/overhangio/tutor/blob/8ccc3588e756178dcf8dc7c0e0b458223720f59d/tutor/templates/apps/openedx/settings/partials/common_all.py#L222-L230) for this.
2. Set up the [codejail service](https://github.com/eduNEXT/codejailservice) directly. What took me so much time reviewing getting a vm with apparmor with tutor etc set up, I think you can skip all that and just run this directly instead!
3. Add the [config from the tutor plugin](https://github.com/eduNEXT/tutor-contrib-codejail/blob/main/tutorcodejail/patches/cms-env), filling in the host name with localhost or wherever you've set it up. 

After that the test should run using the codejail service. You can check this with a print statement or some other way.

Reply:
@xitij2000, would you be able to finalize this PR as the reviewer of [BB-7705](https://tasks.opencraft.com/browse/BB-7705) _(private link)_? I can merge this afterward.

Reply:
Certainly.